src/features/user/Profile/components/RedeemModal.tsx

```diff
@@ -288,69 +327,73 @@ export default function RedeemModal({
-                        value: true,
-                        message: t('redeem:enterRedeemCode'),
-                      },
-                    })}
-                    onChange={(event) => {
-                      console.log(event.target.value);
-                      event.target.value.startsWith('pp.eco/c/')
-                        ? setInputCode(
-                            event.target.value.replace('pp.eco/c/', '')
-                          )
-                        : setInputCode(event.target.value);
-                    }}
-                    value={inputCode}
-                    name={'code'}
-                    placeholder="XAD-1SA-5F1-A"
-                    label=""
-                    variant="outlined"
-                  />
-                </div>
+                {!errorMessage && (
+                  <div className={styles.inputField}>
+                    <MaterialTextField
+                      inputRef={register({
+                        required: {
+                          value: true,
+                          message: t('redeem:enterRedeemCode'),
+                        },
+                      })}
+                      onChange={(event) => {
+                        event.target.value.startsWith('pp.eco/c/')
+                          ? setInputCode(
+                              event.target.value.replace('pp.eco/c/', '')
+                            )
+                          : setInputCode(event.target.value);
+                      }}
+                      value={inputCode}
+                      name={'code'}
+                      placeholder="XAD-1SA-5F1-A"
+                      label=""
+                      variant="outlined"
+                    />
+                  </div>
+                )}
                 {errors.code && (
                   <span className={styles.formErrors}>
                     {errors.code.message}
                   </span>
                 )}
+                {errorMessage && !errors.code && !isUploadingData && (
+                  <span className={styles.formErrors}>{errorMessage}</span>
+                )}
 
                 {errorMessage && (
-                  <span className={styles.formErrors}>{errorMessage}</span>
+                  <button
+                    className="primaryButton redeemAnotherCode"
+                    onClick={handleAnotherCode}
+                  >
+                    {t('redeem:redeemAnotherCode')}
+                  </button>
+                )}
+                {!errorMessage && (
+                  <button
+                    id={'redeemCodeModal'}
+                    onClick={handleSubmit(redeemCode)}
+                    className="primaryButton redeemCode"
+                  >
+                    {isUploadingData ? (
+                      <div className={styles.spinner}></div>
+                    ) : (
+                      t('redeem:redeemCode')
+                    )}
+                  </button>
```

Comment:
Why are we using 2 different classes? It may be better to use a common class in both places and prescribe a padding for the button instead of a max width.